ggml-vulkan.cpp

```diff
@@ -1088,17 +1088,45 @@ void ggml_vk_instance_init() {
 #endif
 
     vk::ApplicationInfo app_info{ "ggml-vulkan", 1, nullptr, 0, VK_API_VERSION };
-    const std::vector<const char*> layers = {
+
+    const std::vector<vk::ExtensionProperties> instance_extensions = vk::enumerateInstanceExtensionProperties();
+#ifdef __APPLE__
```

Comment:
Nit: I would prefer that these get extracted to functions that either do something (in the Apple case) or do nothing (in the other cases) and land at the bottom of this file; see `ggml.c` for an example with `__APPLE__` around line 16583.

Reply:
You're right. I reworked this and the validation layer enable. Can you take a look (and maybe do another sanity check run on Apple)?